File: 633-sum-of-square-numbers/sum-of-square-numbers.cpp
class Solution {
public:
    bool judgeSquareSum(int c) {
        if (c == 0) return true;
        double cutoff = pow(c, 0.5);
        for (int a = 0; a < cutoff; a++){
            double b = pow(c - pow(a, 2), 0.5);
            int int_b = b;
            if (b == int_b){ // suitable choice for b
                return true;
            }
        }

        return false;
    }
};

File: 633-sum-of-square-numbers/sum-of-square-numbers.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        cutoff = int(c ** 0.5)
        for a in range(0, cutoff + 1):
            a_sq = a ** 2
            b = (c - a_sq ) ** 0.5
            if b % 1 == 0:
                return True
        
        return False